lib/elixir/src/elixir_aliases.erl

```diff
@@ -141,4 +144,8 @@ format_error({unloaded_module, Module}) ->
 
 format_error({scheduled_module, Module}) ->
   io_lib:format("module ~ts is not loaded but was defined. This happens because you are trying to use a module in the same context it is defined. Try defining the module outside the context that requires it.",
-    [inspect(Module)]).
\ No newline at end of file
+    [inspect(Module)]);
+
+format_error({circular_module, Module}) ->
+  io_lib:format("module ~ts is not loaded but was defined. This happens because you are trying to use a module in the context of the same module. Try using the module from the context of different module.",
```

Comment:
What about: "you are trying to use/require the same module that is being currently defined. The current module is ~ts"